client/src/components/Indices/filtersMixin.js

```diff
@@ -4,15 +4,40 @@ export default {
     components: {
         IndexFilter,
     },
+    props: {
```

Comment:
Could you refactor the mixin to a composable ?

Reply:
We've talked about this and the mixin itself brings in a lot of other things via the IndexFilter component, refactoring without converting everything to composition API is tricky. Given the original PR started some time ago before composition was an option we should probably keep the mixin for now.

Reply:
Alright, makes it easier for this PR. Further refactoring of this mixin should be pretty separated from the rest of the work in https://github.com/galaxyproject/galaxy/pull/15237 thus independent